server.py

```diff
@@ -0,0 +1,257 @@
+"""Server for Markov chains Tweet generator app"""
+
+from flask import Flask, render_template, request, flash, session, redirect, jsonify
+from flask_login import LoginManager, login_user, logout_user, login_required, current_user
+import jinja2
+
+from model import db, connect_to_db, User, Original_Tweet, UG_Tweet, Author
+import crud
+
+from markovchain.text import MarkovText
+
+import os
+import sys 
+
+app = Flask(__name__)
+app.secret_key = os.environ.get("FLASK_SERVER_KEY")
+
+# Configurations to use Flask Login
+login_manager = LoginManager()
+login_manager.init_app(app)
+login_manager.login_view = 'login'
+
+
+####################################################################
```

Comment:
Same with the model file, you can have a "server" or "routes" folder. These "####" blocks are a good inidicator that you can split these up into separate files

Reply:
Noted! I'm going to leave the routes all together since the app isn't too big yet. But to improve readability, I did separate out the functions that were in the server.py file that weren't related to just serving the pages. 

There's a new markov_helpers.py module to keep the more involved markov algorithms in one place away from the routes - shown in [this commit](https://github.com/missy-davies/chatterbot/commit/ca09075cda7304a6b08ffd958386c331bf6f0d0e).